filament/src/BufferPoolAllocator.h

```diff
@@ -92,7 +92,6 @@ class BufferPoolAllocator {
 
 template<size_t POOL_SIZE, size_t ALIGNMENT, typename AllocatorPolicy, typename LockingPolicy>
 BufferPoolAllocator<POOL_SIZE, ALIGNMENT, AllocatorPolicy, LockingPolicy>::~BufferPoolAllocator() noexcept {
-    assert_invariant(mOutstandingBuffers == mEntries.size());
```

Comment:
@pixelflinger can you think of a way to make this assertion work? Otherwise, we can remove `mOustandingBuffers` since I don't think it's being used for anything.

Reply:
we could count the number of callbacks registered, the number of callbacks that "came back" and check that outstandingBuffer matches the difference?

Not sure if it is worth it.

Reply:
Yeah, I agree